### DISCUSSION TITLE
useWriteContract hook fires every time the component renders

### DISCUSSION THREAD

**Discussion by Javier-Szyfer** [wevm/wagmi#814] [Q&A]
Hi im looking for help with this problem:
Every time the component re renders the hook is  called . I only want to call write or writeAsync when needed.

```
 const { config } = usePrepareContractWrite({
    addressOrName: mainnetZoraAddresses.ReserveAuctionCoreEth,
    contractInterface: ZORA_RESERVE_AUCTION_RINKEBY,
    functionName: "createBid",
    args: [contractAddress, 2],
    overrides: {
      value: bidPrice,
    },
  });

  const { data: createBidData, writeAsync: createBidWrite } = useContractWrite({
    ...config,
    onError(error) {
      toast.error(error.message, {
        toastId: "$s4f24fsrsd45",
      });
      setProcessing(false);
    },
  });

(....some extra code ...)

<button onClick={()=> createBidWrite?.(config)}</button>
```

How can avoid this multiple calls to the hook?
Thanks

**Comment by Irwinhrdz** [comment]
this helped me a lot <br> <br> `... usePrepareContractWrite{ 
... ,
 enabled = flag,
 ...
}`  <br> <br> maybe this Discussion can help you    [#804]( https://github.com/wagmi-dev/wagmi/discussions/804)

**Reply by jxom** [reply]
Yes! The `enabled` flag helps a lot.

**Comment by Harry180** [comment]
You could do it:


`const { write } = useContractWrite(config);`
`<button onClick={createBidWrite}</button>`

It will be called on button click not automatically multiple times in a row.
Please remember that it will throw error if `address` or `abi` is not passed to `usePrepareContractWrite` hook
